### DISCUSSION TITLE
Why does Correlate output "Value" instead of Input like other activities ?

### DISCUSSION THREAD

**Discussion by OvyDev** [elsa-workflows/elsa-core#3197] [Q&A]
Why does Correlate output "Value" instead of Input like other activities ? 
This makes it difficult to use Input from preceding activity from here on. 
It also seem redundant to output "Value" since we can just get CorrelationId afterward.

Is there any other activity that behave like this ?

This is with Elsa v2.8.2
Thanks

**Comment by sfmskywalker** [comment]
When you provide an expression for the `Correlate` activity's `Value` property, you may not know what the correlation ID is  until the expression is evaluated. For example, the expression may be a JS function called `guid()` that generates a new GUID to use as the correlation ID.

For this reason, it may be convenient for the activity to return the actual value as its output.
You are right, you can also use `CorrelationId` afterwards.

> This makes it difficult to use Input from preceding activity from here on.

This is true for pretty much any activity. Input from one activity only flows into the next one, which itself may or may not return output. When output is returned, usually it's something different.

In other words, output of activity A flows as input into activity B. Activity B does not echo back its input - it may or may not produce different output.

So what do you do if activity C needs the output of activity A? One of two things:

1. Access the output of activity A directly from activity C.
2. Store output from activity A in a variable, which then too is accessible from other activities.

Let me know if this makes sense or if I misunderstood your scenario.
